src/api/app/models/package.rb

```diff
@@ -1210,6 +1210,15 @@ def serviceinfo
   def parse_all_history
     answer = source_file('_history')
 
+    cache_revision_commits(doc)
+  end
+
+  def cache_single_commit(revision)
```

Comment:
All 3 methods can now be `cache_revisions(rev: nil)`, the only difference is how you fetch the `_history`.